be/src/exec/pipeline/exchange/sink_buffer.cpp

```diff
@@ -335,10 +309,15 @@ Status SinkBuffer::_try_to_send_rpc(const TUniqueId& instance_id, const std::fun
                     return Status::OK();
                 }
                 // this is the last eos query, set query stats
-                _eos_query_stats[instance_id.lo]->merge_pb(request.params->query_statistics());
-                request.params->clear_query_statistics();
-                _eos_query_stats[instance_id.lo]->to_pb(request.params->mutable_query_statistics());
-                _eos_query_stats[instance_id.lo]->clear();
+                if (auto final_stats = _fragment_ctx->runtime_state()->query_ctx()->intermediate_query_statistic()) {
```

Comment:
what if there is a short circuit, the receiver side won't wait the last eos packet.

Reply:
it is up to the sent stats by `_sent_audit_stats_frequency `